core/src/main/scala/magnolia1/monadic.scala

```diff
@@ -0,0 +1,56 @@
+package magnolia1
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Success, Try}
+
+trait Functor[F[_]] {
+  type Apply[X] = F[X]
+  def point[A](value: A): F[A]
+  def map[A, B](from: F[A])(fn: A => B): F[B]
+}
+
+trait Monadic[F[_]] extends Functor[F] {
+  def flatMap[A, B](from: F[A])(fn: A => F[B]): F[B]
+}
+
+object Monadic {
```

Comment:
I assume this is copied from mercator, we should add a comment pointing back to the original source

Reply:
As I wrote in commit msg it was copied from scala3 branch so it depends on from where it was copied there

Reply:
Ah :D I didn't read the commit msg, that's fine then